R/checkers.R

```diff
@@ -0,0 +1,207 @@
+#' Check if `<data.frame>` defining age-stratified hospitalisation or death risk
+#' is correct
+#'
+#' @param x A `<data.frame>`.
+#'
+#' @return A `<data.frame>`, also called for error side-effects when input is
+#' invalid.
+#' @keywords internal
+.check_risk_df <- function(x, age_range) {
+  # check input
+  stopifnot(
+    "column names should be 'age_limit' & 'risk'" =
+      setequal(c("age_limit", "risk"), colnames(x)),
+    "minimum age of lowest age group should match lower age range" =
+      age_range[1] == min(x$age_limit),
+    "lower bound of oldest age group must be lower than highest age range" =
+      age_range[2] > max(x$age_limit),
+    "age limit or risk cannot be NA or NaN" =
+      !anyNA(x),
+    "risk should be between 0 and 1" =
+      min(x$risk) >= 0 && max(x$risk) <= 1,
+    "age limit in risk data frame must be unique" =
+      anyDuplicated(x$age_limit) == 0
+  )
+
+  # order risk df on age_limit
+  x <- x[order(x$age_limit), ]
+
+  # format risk data frame
+  age_range_ <- age_range[1]:age_range[2]
+  # findInterval inclusive/exclusive bound rules match age bracket
+  age_groups <- unname(split(age_range_, findInterval(age_range_, x$age_limit)))
```

Comment:
Nice use of `findInterval`, a function I didn't know of - coded a `Position()` call recently elsewhere that could be replaced with this